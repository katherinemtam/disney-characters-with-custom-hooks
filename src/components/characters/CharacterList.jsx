import React, { useState } from 'react';
import { useCharacters } from '../../state/character';
import { Link } from 'react-router-dom';
import Character from './Character';

const CharacterList = () => {
  const [page, setPage] = useState(1);
  const { characters, loading } = useCharacters(page);

  const characterElements = characters.map((character) => (
    <li key={character._id}>
      <Link to={`/${character._id}`}>
        <Character {...character} />
      </Link>
    </li>
  ));

  if (loading) return <h1>Loading...</h1>;
  return (
    <>
      <button
        disabled={page <= 1}
        onClick={() => setPage((prevPage) => prevPage - 1)}
      >
        &lt;
      </button>
      Page {page}
      <button onClick={() => setPage((prevPage) => prevPage + 1)}>&gt;</button>
      <ul>{characterElements}</ul>
    </>
  );
};

export default CharacterList;
