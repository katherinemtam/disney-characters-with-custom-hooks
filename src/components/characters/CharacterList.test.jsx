import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CharacterList from './CharacterList';

describe('CharacterList component', () => {
  it('renders CharacterList', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
