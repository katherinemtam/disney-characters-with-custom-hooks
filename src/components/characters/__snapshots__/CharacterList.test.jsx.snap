// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <h1>
    Loading...
  </h1>
</DocumentFragment>
`;
